### PR TITLE
feat(server): add IP trust list for reverse proxy

### PR DESCRIPTION
**Pull request by hitech95**
Added support for configurable trusted proxy IPs using environment variables. This enhancement allows you to specify trusted proxy IPs directly through environment settings, improving flexibility and security in various deployment environments.

It follow the rules of express `trust proxy` option. See [Behind Proxies](https://expressjs.com/en/guide/behind-proxies.html)

New Environment Variable:

    IMMICH_TRUSTED_PROXIES: A comma-separated list of IP addresses that should be trusted as proxies.

Example Usage in Docker Compose:

```yaml

version: '3.8'

immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
   environment:
      - TRUSTED_PROXY_IPS=192.168.0.1,192.168.0.2
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always
```

With this fix, you can now manage trusted proxy IPs without modifying the application code, making it easier to adjust settings in different environments.

### CODE REVIEW

**Comment by bo0tzz** [comment]
Please also add this to the [env var validation](https://github.com/immich-app/immich/blob/b53bd8c525c9760871529334b26571f672eabaf0/server/src/config.ts#L374) and [the docs](https://github.com/immich-app/immich/blob/b53bd8c525c9760871529334b26571f672eabaf0/docs/docs/install/environment-variables.md?plain=1#L39).

File: server/src/workers/api.ts [hunk 1]
```diff
@@ -14,9 +14,18 @@ import { useSwagger } from 'src/utils/misc';
 
 const host = process.env.HOST;
 
+function parseTrustedProxy(input?: string) {
+  if (input === undefined || input === '') {
```
Comment (danieldietzler) [inline server/src/workers/api.ts:18]:
```suggestion
  if (!input) {
```

**Comment by hitech95** [comment]
> Please also add this to the [env var validation].

I cannot validate the IPs without creating a huge regexp, can I just validate the string?
accepting suggestions for that.

Those should all be valid inputs and validating them precisely is a bit of a mess.

```
console.log(validateAndSplit("192.168.1.1;2001:db8::ff00:42:8329;10.0.0.0/24")); // ["192.168.1.1", "2001:db8::ff00:42:8329", "10.0.0.0/24"]
console.log(validateAndSplit("")); // []
console.log(validateAndSplit(undefined)); // []
```

File: server/src/workers/api.ts (hunk 1 above)
Comment (hitech95) [reply server/src/workers/api.ts:18]: Ok, you prefere a less explicit check :D

Comment (danieldietzler) [reply server/src/workers/api.ts]: I think it's equally explicit :P
If you prefer keeping it I'm also fine with that though

Comment (hitech95) [reply server/src/workers/api.ts]: The end result is the same, I think that the explicit check is easier to see as not everyone knows that empty string is false.
I've already applied the change anyway 😄

**Review by danieldietzler — APPROVED** [review]
LGTM

**Review by bo0tzz — APPROVED** [review]

**Pull request merged**